apps/@sparrow-desktop/src/containers/api/api.common.ts

```diff
@@ -539,8 +546,78 @@ const connectWebSocket = async (
         });
         notifications.success("WebSocket connected successfully.");
 
+        const errListner = await once(`ws_error_${tabId}`, (event) => {
+          webSocketDataStore.update((webSocketDataMap) => {
+            const wsData = webSocketDataMap.get(tabId);
+
+            if (wsData) {
+              let msgListner: any = wsData.msgListner;
+
+              wsData.messages.unshift({
+                data: `Disconnected from ${url}`,
+
+                transmitter: "disconnector",
+
+                timestamp: formatTime(new Date()),
+
+                uuid: uuidv4(),
+              });
+
+              wsData.status = "disconnected";
+
+              webSocketDataMap.set(tabId, wsData);
+
+              if (msgListner) {
+                msgListner();
+              }
+            }
+
+            return webSocketDataMap;
+          });
+
+          notifications.error("Connection Lost");
+
+          console.log("abrupt disconnection");
+        });
+
+        const closeListen = await once(
+          `ws_graceful_disconnect_${tabId}`,
+          (event) => {
+            webSocketDataStore.update((webSocketDataMap) => {
+              const wsData = webSocketDataMap.get(tabId);
+
+              if (wsData) {
+                let msgListner: any = wsData.msgListner;
+
+                wsData.messages.unshift({
+                  data: `Disconnected from ${url}`,
+
+                  transmitter: "disconnector",
+
+                  timestamp: formatTime(new Date()),
+
```

Comment:
these two listeners should be not required, should be only handled with current listener using parameters.

Reply:
Understood. Should I create a new pull request, or would you prefer I push the changes directly?
@itsmdasifraza